worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/lowlevel/StagingQueueTargetQueuePair.java

```diff
@@ -86,42 +146,58 @@ public void handleStagedMessage(final String consumerTag, final Envelope envelop
                     .priority(properties.getPriority())
                     .build();
 
-            //Hack the To See https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=614206
-            final JsonObject jsonObject =
-                    gson.fromJson(new String(body, StandardCharsets.UTF_8), JsonObject.class);
-            jsonObject.addProperty("to", targetQueue.getName());
-            final String s = gson.toJson(jsonObject);
-            //End Hack
+            targetQueueChannel.basicPublish("", targetQueue.getName(), basicProperties, body);
+        } catch (final IOException basicPublishException) {
 
-            targetQueueChannel.basicPublish("", targetQueue.getName(), basicProperties, 
-                    s.getBytes(StandardCharsets.UTF_8));
-        }
-        catch (final IOException e) {
-            LOGGER.error("Exception publishing to '{}' {}", targetQueue.getName(),
-                    e.toString());
-            stagingQueueChannel.basicCancel(consumerTag);
+            LOGGER.error("Exception publishing to '{}' for StagingQueueTargetQueuePair '{}' {}",
+                    targetQueue.getName(), this, basicPublishException.toString());
 
+            cancelConsumer(consumerTag);
```

Comment:
We're cancelling the consumer here, so any future delveries will be rejected. 
I think we should be returning after this and not bothering to messageCount.incrementAndGet(); and potentially calling cancelConsumer again. 

What happens the message from the staging queue that was just delivered? It seems like we have to wait for https://www.rabbitmq.com/consumers.html#acknowledgement-timeout

It seems that once the the channel is closed, all unnack'd messages will be requeued.
https://www.rabbitmq.com/consumers.html#unsubscribing
_All previously unconfirmed deliveries will not be affected in any way. To re-queue in-flight deliveries, the application must close the channel._

Reply:
I have updated this so that in the case where we cancel the consumer due to an error, we also close the channel, which allows unacked messages to to be immediately required.

I've added more detail on this in the comment at the start of the `handleStagedMessage` method.